Dockerfile

```diff
@@ -2,14 +2,24 @@
 # Build Stage
 # ------------------------------------------------------------------------------
 
-FROM alpine:3.16 as limitador-build
+FROM registry.access.redhat.com/ubi8/ubi:8.7 as limitador-build
 ENV CARGO_NET_GIT_FETCH_WITH_CLI=true
 
 ARG RUSTC_VERSION=1.67.1
-RUN apk update \
-    && apk upgrade \
-    && apk add build-base binutils-gold openssl3-dev protoc protobuf-dev curl git \
-    && curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- --no-modify-path --profile minimal --default-toolchain ${RUSTC_VERSION} -c rustfmt -y
+
+# the powertools repo is required for protobuf-c and protobuf-devel
+RUN dnf -y --setopt=install_weak_deps=False --setopt=tsflags=nodocs install \
+      http://mirror.centos.org/centos/8-stream/BaseOS/`arch`/os/Packages/centos-gpg-keys-8-6.el8.noarch.rpm \
```

Comment:
centos repos?  I thought it reached EOL

Would it work compiling in alpine and run in ubi8?

Reply:
centos stream is fully supported until at least next year: https://wiki.centos.org/About/Product

Reply:
>Would it work compiling in alpine and run in ubi8?

Why would you prefer that? Just to avoid the CentOS repo?

Reply:
Well, the build stage seems now harder to maintain. With alpine, it was easier to maintain (not simpler though).

Not a big deal, though. LGTM

Reply:
And yes, CENTOS in my mind is an echo of the PAST

Reply:
> Well, the build stage seems now harder to maintain. With alpine, it was easier to maintain (not simpler though).

definitely harder / more obscure to maintain. 👍🏼 

the alternative would be to build protobuf from source though, which I guess is worse... With that assumption, I'll go ahead and resolve - but please reopen if you prefer us to move in that direction.